cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerDecomissioner.java

```diff
@@ -196,9 +196,9 @@ private void addInstancesNotKnownByCMWithLimit(Integer limit, Set<InstanceMetaDa
         if (hostsToRemove.size() != hostNames.size()) {
             LOGGER.debug("Not all hosts found in the given host group. [{}, {}]", hostGroup.getName(), hostNames);
         }
-        ClustersResourceApi clustersResourceApi = clouderaManagerApiFactory.getClustersResourceApi(client);
+        HostsResourceApi hostsResourceApi = clouderaManagerApiFactory.getHostsResourceApi(client);
         try {
-            ApiHostList hostRefList = clustersResourceApi.listHosts(stack.getName(), null, null, null);
+            ApiHostList hostRefList = hostsResourceApi.readHosts(null, null, null);
```

Comment:
what happens if third param (view) is null? Will it return with full view? Or with simplest view?

Reply:
What happens if CB sends a node to be decommissioned when it is not part of CM cluster?

Reply:
it seems it is handled here in the filter:
```
Sets.newHashSet(hostsToRemove.keySet()).stream()
                    .filter(hostName -> !runningHosts.contains(hostName))
                    .forEach(hostsToRemove::remove);
```

Reply:
I set it to SUMMARY just to be sure.

Reply:
Decommission does not seem to be cluster based. It was able to decommission it with no problem.